homer0/projext-plugin-rollup
# fix(project): fix code splitting issue with core-js

### What does this PR do?

This is a follow up for #.

The package `core-js` has a module that does `var location = `, and when doing code splitting, because Rollup is not scoping the module, the browser (for some reason) is taking it as a redirection to the same page... causing a redirection loop.

The solution was to create a plugin (like everything for Rollup at this point) that would wrap `rollup-plugin-replace` and fix issues on specific modules:

```js
moduleReplace({
  instructions: [{
    module: /node_modules\/core-js\/internals\/task\.js$/i,
    search: /(\s*)location(\.|\s*=)/i,
    replace: '$1_location$2',
  }],
});
```

It basically transforms `location` into `_location`. Yes, I could've used the `rollup-plugin-replace`, but if this happens again, the engine now has a centralized place to apply these fixes.

### How should it be tested manually?

In order to test this, you need to install `homer0/projext#next` and make sure you delete the copy of `projext` `npm`/`yarn` will install inside this package's `node_modules`.

Try running a target with code splitting, it should work now.

And, of course...

```bash
yarn test
# or
npm test
```
